core/src/test/java/com/gentics/mesh/core/webroot/WebRootEndpointTest.java

```diff
@@ -94,6 +103,46 @@ public void testReadBinaryNode() throws IOException {
 		assertNotNull(downloadResponse);
 	}
 
+	@Test
+	public void testReadS3BinaryNode() {
+		HibNode node = content("news_2015");
+
+		try (Tx tx = tx()) {
+			SchemaVersionModel schema = schemaContainer("content").getLatestVersion().getSchema();
+			S3BinaryFieldSchema s3BinaryFieldSchema = new S3BinaryFieldSchemaImpl();
+			s3BinaryFieldSchema.setName("s3");
+			s3BinaryFieldSchema.setLabel("Some label");
+			schema.addField(s3BinaryFieldSchema);
+			schema.setSegmentField("s3");
+			schemaContainer("content").getLatestVersion().setSchema(schema);
+			tx.success();
+		}
+		String uploadBody = "{\n" +
+				"\t\"language\":\"en\",\n" +
+				"\t\"version\":\"1.0\",\n" +
+				"\t\"filename\":\"test.jpg\"\n" +
+				"}";
+
+		// 2. Update the binary data
+		call(() -> client().updateNodeS3BinaryField(
+				PROJECT_NAME,
+				node.getUuid(),
```

Comment:
This part crashes - DAOs are lazy and require DB transaction for the fields retrieval. Please consider using `String nodeUuid = tx(() -> node.getUuid());` approach, as in the tests above.